src/Settings/DefaultPathManager.cpp

```diff
@@ -26,8 +26,7 @@ QMap<QString, QString> DefaultPathManager::defaultPath;
 
 QString DefaultPathManager::defaultPathForAction(const QString &action)
 {
-    const auto result =
-        convertPath(SettingsManager::get(QString("Default Path/Action/%1/Uses").arg(action)).toString());
+    auto result = convertPath(SettingsManager::get(QString("Default Path/Action/%1/Uses").arg(action)).toString());
```

Comment:
returning const result prevent compiler from optimising return value.

Reply:
I think `const` improves readability :thinking: And Qt has Implicit Sharing.

Reply:
The reason why this comes out as performance issue because usually compiler tries to move the return value to caller so no copy is made for returned value. If you make return variable `const` the compiler cannot perform move and is forced to return the value by copy. It is called return value optimization.

Reply:
> The reason why this comes out as performance issue because usually compiler tries to move the return value to caller so no copy is made for returned value. If you make return variable `const` the compiler cannot perform move and is forced to return the value by copy. It is called return value optimization.

I didn't ask you what is "return value optimization". I said: "const improves readability and Qt has Implicit Sharing".

I'd say I have wasted more time handling this issue than the "performance improvement" applied to even millions of users.

Reply:
I'll disable this check.